ert_shared/models/base_run_model.py

```diff
@@ -205,45 +206,62 @@ def is_forward_model_finished(progress):
+            queue_index = run_arg.getQueueIndex()
+        except ValueError:
+            return
+
+        status = None
+        if self._job_queue:
+            status = self._job_queue.getJobStatus(queue_index)
+
+        if status in [
+                JobStatusType.JOB_QUEUE_PENDING,
+                JobStatusType.JOB_QUEUE_SUBMITTED,
+                JobStatusType.JOB_QUEUE_WAITING
+                ]:
+            return
+
+        fms = self.realization_progress[iteration].get(run_arg.iens, None)
+
+        #Dont load from file if you are finished
+        if fms and BaseRunModel.is_forward_model_finished(fms[0]):
+            jobs = self.realization_progress[iteration][run_arg.iens][0]
+        else:
+            fms = ForwardModelStatus.load(run_arg.runpath, num_retry=1)
+            if not fms:
+                return
+
+            jobs = fms.jobs
+        self.realization_progress[iteration][run_arg.iens] = jobs, status
+
+
     def updateDetailedProgress(self):
         if not self._run_context:
             return
 
         iteration = self._run_context.get_iter()
         if iteration not in self.realization_progress:
             self.realization_progress[iteration] = {}
-        for idx, run_arg in enumerate(self._run_context):
-            if not self._run_context.is_active(idx):
-                continue
-            try:
-                # will throw if not yet submitted (is in a limbo state)
-                queue_index = run_arg.getQueueIndex()
-            except ValueError:
-                continue
-
-            status = None
-            if self._job_queue:
-                status = self._job_queue.getJobStatus(queue_index)
-
-            if status in [
-                    JobStatusType.JOB_QUEUE_PENDING,
-                    JobStatusType.JOB_QUEUE_SUBMITTED,
-                    JobStatusType.JOB_QUEUE_WAITING
-                    ]:
-                continue
-
-            fms = self.realization_progress[iteration].get(run_arg.iens, None)
-
-            #Dont load from file if you are finished
-            if fms and BaseRunModel.is_forward_model_finished(fms[0]):
-                jobs = self.realization_progress[iteration][run_arg.iens][0]
-            else:
-                fms = ForwardModelStatus.load(run_arg.runpath, num_retry=1)
-                if not fms:
-                    continue
 
-                jobs = fms.jobs
-            self.realization_progress[iteration][run_arg.iens] = jobs, status
+        try:
+            # Run context might be set to None by concurrent threads,
+            # which will results in an Attribute Error
+            for idx, run_arg in enumerate(self._run_context):
+                self.update_progress_for_index(iteration, idx, run_arg)
+        except AttributeError as e:
+            if self._run_context is None:
+                logging.debug(
+                    "Ignoring exception in run model (run_context is None): {}".format(
```

Comment:
Was just wondering shall we also log `iteration` and `idx` or can this be deduced?

Reply:
The interesting thing here is that depending on when in the code run_context is set to None, it's not necessarily certain that idx will be set. I also think the value of knowing exactly what index and iteration it was updating when run_context was set to None is limited, because it's not that specific index that caused any error, it just happened to update the status of it :slightly_smiling_face:

Reply:
I see. Definitely an event driven messaging would help, something like this perhaps: 
https://github.com/Numergy/signalslot

Reply:
Yes, that would be a more viable, long term solution!